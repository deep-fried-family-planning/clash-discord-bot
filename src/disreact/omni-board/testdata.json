{
  "app_permissions"               : "2251799813685247",
  "application_id"                : "1256809603715436606",
  "authorizing_integration_owners": {"0": "1287829383544963154"},
  "channel"                       : {
    "flags"              : 0,
    "guild_id"           : "1287829383544963154",
    "id"                 : "1317321691809972254",
    "last_message_id"    : "1333657395372756992",
    "member_count"       : 2,
    "member_ids_preview" : ["267835119421751306", "644290645350940692"],
    "message_count"      : 114,
    "name"               : "dev omni",
    "owner_id"           : "644290645350940692",
    "parent_id"          : "1298154982641500182",
    "permissions"        : "2251799813685247",
    "rate_limit_per_user": 0,
    "thread_metadata"    : {
      "archive_timestamp"    : "2024-12-14T02:46:05.981000+00:00",
      "archived"             : false,
      "auto_archive_duration": 4320,
      "create_timestamp"     : "2024-12-14T02:46:05.981000+00:00",
      "locked"               : false
    },
    "total_message_sent" : 146,
    "type"               : 11
  },
  "channel_id"                    : "1317321691809972254",
  "context"                       : 0,
  "data"                          : {"component_type": 2, "custom_id": "components:1:button:0"},
  "entitlement_sku_ids"           : [],
  "entitlements"                  : [],
  "guild"                         : {
    "features": ["NEWS", "COMMUNITY", "AUTO_MODERATION"], "id": "1287829383544963154", "locale": "en-US"
  },
  "guild_id"                      : "1287829383544963154",
  "guild_locale"                  : "en-US",
  "id"                            : "1333657404356820992",
  "locale"                        : "en-US",
  "member"                        : {
    "avatar"                      : null,
    "banner"                      : null,
    "communication_disabled_until": null,
    "deaf"                        : false,
    "flags"                       : 0,
    "joined_at"                   : "2024-09-23T17:34:11.807000+00:00",
    "mute"                        : false,
    "nick"                        : null,
    "pending"                     : false,
    "permissions"                 : "2251799813685247",
    "premium_since"               : null,
    "roles"                       : ["1298138273478086689"],
    "unusual_dm_activity_until"   : null,
    "user"                        : {
      "avatar"                : "7ad8c035b5caa70e2132e550aba50fcf",
      "avatar_decoration_data": null,
      "clan"                  : null,
      "discriminator"         : "0",
      "global_name"           : "NotStr8DontH8",
      "id"                    : "644290645350940692",
      "primary_guild"         : null,
      "public_flags"          : 4194304,
      "username"              : "yourguyryry"
    }
  },
  "message"                       : {
    "application_id"         : "1256809603715436606",
    "attachments"            : [],
    "author"                 : {
      "avatar"                : "5f71735406403cdc51158038e24bb671",
      "avatar_decoration_data": null,
      "bot"                   : true,
      "clan"                  : null,
      "discriminator"         : "5344",
      "global_name"           : null,
      "id"                    : "1256809603715436606",
      "primary_guild"         : null,
      "public_flags"          : 524288,
      "username"              : "QualDeepFryer"
    },
    "channel_id"             : "1317321691809972254",
    "components"             : [
      {
        "components": [{"custom_id": "components:1:button:0", "id": 2, "label": "Link", "style": 3, "type": 2}],
        "id"        : 1,
        "type"      : 1
      }
    ],
    "content"                : "",
    "edited_timestamp"       : null,
    "embeds"                 : [
      {
        "content_scan_version": 1, "description": "OmniStart", "image": {
        "flags"    : 0,
        "height"   : 0,
        "proxy_url": "https://images-ext-1.discordapp.net/external/_CXbNDbKdviGWXOKuEUWO1fAFgXYmknPNAXV6OSWYQI/https/dffp.org/dr/OmniStart/OmniStart/-/-",
        "url"      : "https://dffp.org/dr/OmniStart/OmniStart/-/-/-",
        "width"    : 0
      }, "title"              : "OmniStart", "type": "rich"
      }
    ],
    "flags"                  : 0,
    "id"                     : "1333657395372756992",
    "interaction"            : {
      "id": "1333657393770528859", "name": "smoke", "type": 2, "user": {
        "avatar"                : "7ad8c035b5caa70e2132e550aba50fcf",
        "avatar_decoration_data": null,
        "clan"                  : null,
        "discriminator"         : "0",
        "global_name"           : "NotStr8DontH8",
        "id"                    : "644290645350940692",
        "primary_guild"         : null,
        "public_flags"          : 4194304,
        "username"              : "yourguyryry"
      }
    },
    "interaction_metadata"   : {
      "authorizing_integration_owners": {"0": "1287829383544963154"},
      "command_type"                  : 1,
      "id"                            : "1333657393770528859",
      "name"                          : "smoke",
      "type"                          : 2,
      "user"                          : {
        "avatar"                : "7ad8c035b5caa70e2132e550aba50fcf",
        "avatar_decoration_data": null,
        "clan"                  : null,
        "discriminator"         : "0",
        "global_name"           : "NotStr8DontH8",
        "id"                    : "644290645350940692",
        "primary_guild"         : null,
        "public_flags"          : 4194304,
        "username"              : "yourguyryry"
      }
    },
    "mention_everyone"       : false,
    "mention_roles"          : [],
    "mentions"               : [],
    "pinned"                 : false,
    "position"               : 145,
    "timestamp"              : "2025-01-28T04:38:21.213000+00:00",
    "tts"                    : false,
    "type"                   : 20,
    "webhook_id"             : "1256809603715436606"
  },
  "token"                         : "aW50ZXJhY3Rpb246MTMzMzY1NzQwNDM1NjgyMDk5Mjo3OHpLcjM1S2ZmdFhuVWsyRTNqb0FWZlRHSXpxbDNFWjNWWWdGSDRMRGNCRk1DOTBzbU5pMU54ZHFJWEdSaE0wbkU3aFNVaG03ZmhmWlNlaWpTbmV5elRuSG5GWElvR2prek9aRERHcGt4cTU3U0d6b2xVWnR6ZjNRT2NGUUdsZg",
  "type"                          : 3,
  "version"                       : 1
}
