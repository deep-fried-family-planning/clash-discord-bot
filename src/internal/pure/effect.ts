import {Order} from 'effect';


export {
    Effect as E,
    Console as CSL,
    Data as D,
    DateTime as DT,
    Redacted as RDT,
    Config as CFG,
    Context as CTX,
    Layer as L,
    Cache as C,
    Logger as Logger,
    LogLevel,
    LogSpan,
    Schema as S,
    Brand as B,
    RateLimiter as RL,
    Metric as Metric,
    Order as ORD,
    Option as O,
    Equal as EQ,

    Cron,

    pipe,
    pipe as p,
    flow,
    flow as f,

} from 'effect';


export {
    gen as g,

} from 'effect/Effect';


export const ORDS = Order.string;
export const ORDSR = Order.reverse(Order.string);
export const ORDN = Order.number;
export const ORDNR = Order.reverse(Order.number);


export * as Ar from 'effect/Array'; // collection array
export * as Kv from 'effect/Record'; // collection key-value
