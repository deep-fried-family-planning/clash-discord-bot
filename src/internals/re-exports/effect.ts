export {
    Effect as E,

    Config as Cfg,
    Redacted as Rdt,

    Console as CSL,

    Data as D,
    DateTime as DT,

    Redacted as RDT,
    Config as CFG,
    Context as CTX,
    Layer as L,

    Cache as C,

    Logger as Logger,
    LogLevel,
    LogSpan,

    Schema as S,

    Brand as B,

    RateLimiter as RL,

    Metric as Metric,

    flow,
    pipe,

} from 'effect';
