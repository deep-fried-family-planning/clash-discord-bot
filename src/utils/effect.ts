export {
    Effect as E,

    Config as Cfg,
    Redacted as Rdt,

    Layer as L,

    Logger as Logger,
    LogLevel,
    LogSpan,

    RateLimiter as RL,

    Metric as Metric,

    flow,
    pipe,

} from 'effect';
