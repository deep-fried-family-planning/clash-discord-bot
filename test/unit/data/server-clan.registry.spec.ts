import {ServerClanRegistry} from '#src/data/index.ts';
import {mockCoc, mockCocLayer} from '#unit/data/mock-coc.ts';
import {TestDataServer, TestDataServerClanElderVerified, TestDataServerClanLeaderVerified, TestDataUser, TestDataUserPlayer} from '#unit/data/mock-db.testdata.ts';
import {mockDb, mockDbLayer} from '#unit/data/mock-db.ts';
import {it} from '@effect/vitest';
import * as E from 'effect/Effect';
import {pipe} from 'effect/Function';

it.effect('when registering a new server clan', E.fn(function* () {
  mockDb.get
    .mockReturnValueOnce(E.succeed({Item: TestDataServer}));

  mockDb.query
    .mockReturnValueOnce(E.succeed({Items: [TestDataUser, TestDataUserPlayer]}))
    .mockReturnValueOnce(E.succeed({Items: []}));

  mockCoc.getClan
    .mockReturnValueOnce(E.succeed({
      tag        : 'clan',
      name       : 'ClanName',
      description: 'ClanDescription',
      members    : [{
        tag : 'player',
        role: 'leader',
      }],
    }));

  const actual = yield* pipe(
    ServerClanRegistry.register({
      caller_id   : 'user',
      caller_roles: ['admin'],
      guild_id    : 'guild',
      clan_tag    : 'clan',
      payload     : {
        countdown: 'countdown',
      },
    }),
    E.provide(mockDbLayer),
    E.provide(mockCocLayer),
  );

  expect(actual).toMatchInlineSnapshot(`
    {
      "description": "Success",
    }
  `);
  expect(mockCoc.getClan).toBeCalled();
  expect(mockCoc.getClan.mock.calls[0][0]).toMatchSnapshot();
  expect(mockDb.get).toBeCalled();
  expect(mockDb.get.mock.calls[0][0].Key).toMatchSnapshot();
  expect(mockDb.query).toBeCalled();
  expect(mockDb.query.mock.calls[0][0].KeyConditionExpression).toMatchSnapshot();
  expect(mockDb.query.mock.calls[1][0].KeyConditionExpression).toMatchSnapshot();
  expect(mockDb.put).toBeCalled();
  expect(mockDb.put.mock.calls[0][0].Item).toMatchSnapshot();
  expect(mockDb.delete).not.toBeCalled();
}));

describe('given caller user is not registered', () => {
  it.effect('when registering', E.fn(function* () {
    mockDb.get
      .mockReturnValueOnce(E.succeed({Item: TestDataServer}));

    mockDb.query
      .mockReturnValueOnce(E.succeed({Items: [TestDataUserPlayer]}))
      .mockReturnValueOnce(E.succeed({Items: []}));

    mockCoc.getClan
      .mockReturnValueOnce(E.succeed({
        tag        : 'clan',
        name       : 'ClanName',
        description: 'ClanDescription',
        members    : [{
          tag : 'player',
          role: 'leader',
        }],
      }));

    const actual = yield* pipe(
      ServerClanRegistry.register({
        caller_id   : 'user',
        caller_roles: ['admin'],
        guild_id    : 'guild',
        clan_tag    : 'clan',
        payload     : {
          countdown: 'countdown',
        },
      }),
      E.provide(mockDbLayer),
      E.provide(mockCocLayer),
      E.catchAll((cause) => E.succeed(cause)),
    );

    expect(actual).toMatchInlineSnapshot(`[RegistryAdminError: Your Discord user account is not registered.]`);
    expect(mockCoc.getClan).not.toBeCalled();
    expect(mockDb.get).toBeCalled();
    expect(mockDb.get.mock.calls[0][0].Key).toMatchSnapshot();
    expect(mockDb.query).toBeCalledTimes(1);
    expect(mockDb.query.mock.calls[0][0].KeyConditionExpression).toMatchSnapshot();
    expect(mockDb.put).not.toBeCalled();
    expect(mockDb.delete).not.toBeCalled();
  }));
});

describe('given caller server is not registered', () => {
  it.effect('when registering', E.fn(function* () {
    const actual = yield* pipe(
      ServerClanRegistry.register({
        caller_id   : 'user',
        caller_roles: ['admin'],
        guild_id    : 'guild',
        clan_tag    : 'clan',
        payload     : {
          countdown: 'countdown',
        },
      }),
      E.provide(mockDbLayer),
      E.provide(mockCocLayer),
      E.catchAll((cause) => E.succeed(cause)),
    );
  }));
});

describe('given caller is not admin', () => {
  it.effect('when registering', E.fn(function* () {
    mockDb.get
      .mockReturnValueOnce(E.succeed({Item: TestDataServer}));

    mockDb.query
      .mockReturnValueOnce(E.succeed({Items: [TestDataUser, TestDataUserPlayer]}))
      .mockReturnValueOnce(E.succeed({Items: []}));

    mockCoc.getClan
      .mockReturnValueOnce(E.succeed({
        tag        : 'clan',
        name       : 'ClanName',
        description: 'ClanDescription',
        members    : [{
          tag : 'player',
          role: 'leader',
        }],
      }));

    const actual = yield* pipe(
      ServerClanRegistry.register({
        caller_id   : 'user',
        caller_roles: [],
        guild_id    : 'guild',
        clan_tag    : 'clan',
        payload     : {
          countdown: 'countdown',
        },
      }),
      E.provide(mockDbLayer),
      E.provide(mockCocLayer),
      E.catchAll((cause) => E.succeed(cause)),
    );

    expect(actual).toMatchInlineSnapshot(`[RegistryAdminError: You are not authorized to register a clan for this server.]`);
    expect(mockCoc.getClan).not.toBeCalled();
    expect(mockDb.get).toBeCalled();
    expect(mockDb.get.mock.calls[0][0].Key).toMatchSnapshot();
    expect(mockDb.query).not.toBeCalled();
    expect(mockDb.put).not.toBeCalled();
    expect(mockDb.delete).not.toBeCalled();
  }));
});

describe('given caller has no verified accounts in clan', () => {
  it.effect('when registering', E.fn(function* () {
    mockDb.get
      .mockReturnValueOnce(E.succeed({Item: TestDataServer}));

    mockDb.query
      .mockReturnValueOnce(E.succeed({Items: [TestDataUser, TestDataUserPlayer]}))
      .mockReturnValueOnce(E.succeed({Items: []}));

    mockCoc.getClan
      .mockReturnValueOnce(E.succeed({
        tag        : 'clan',
        name       : 'ClanName',
        description: 'ClanDescription',
        members    : [],
      }));

    const actual = yield* pipe(
      ServerClanRegistry.register({
        caller_id   : 'user',
        caller_roles: ['admin'],
        guild_id    : 'guild',
        clan_tag    : 'clan',
        payload     : {
          countdown: 'countdown',
        },
      }),
      E.provide(mockDbLayer),
      E.provide(mockCocLayer),
      E.catchAll((cause) => E.succeed(cause)),
    );

    expect(actual).toMatchInlineSnapshot(`[RegistryAdminError: Your Discord account has no verified player links (elder or above) in this clan.]`);
    expect(mockCoc.getClan).toBeCalledTimes(1);
    expect(mockCoc.getClan.mock.calls[0][0]).toMatchSnapshot();
    expect(mockDb.get).toBeCalledTimes(1);
    expect(mockDb.get.mock.calls[0][0].Key).toMatchSnapshot();
    expect(mockDb.query).toBeCalledTimes(2);
    expect(mockDb.query.mock.calls[0][0].KeyConditionExpression).toMatchSnapshot();
    expect(mockDb.query.mock.calls[1][0].KeyConditionExpression).toMatchSnapshot();
    expect(mockDb.put).not.toBeCalled();
    expect(mockDb.delete).not.toBeCalled();
  }));
});

describe('given server clan is already registered with elder verification to a different server', () => {
  it.effect('when registering', E.fn(function* () {
    mockDb.get
      .mockReturnValueOnce(E.succeed({Item: TestDataServer}));

    mockDb.query
      .mockReturnValueOnce(E.succeed({Items: [TestDataUser, TestDataUserPlayer]}))
      .mockReturnValueOnce(E.succeed({Items: [TestDataServerClanElderVerified]}));

    mockCoc.getClan
      .mockReturnValueOnce(E.succeed({
        tag        : 'clan',
        name       : 'ClanName',
        description: 'ClanDescription',
        members    : [{
          tag : 'player',
          role: 'leader',
        }],
      }));

    const actual = yield* pipe(
      ServerClanRegistry.register({
        caller_id   : 'user',
        caller_roles: ['admin'],
        guild_id    : 'guild',
        clan_tag    : 'clan',
        payload     : {
          countdown: 'countdown',
        },
      }),
      E.provide(mockDbLayer),
      E.provide(mockCocLayer),
    );

    expect(actual).toMatchInlineSnapshot(`
      {
        "description": "Success",
      }
    `);
    expect(mockCoc.getClan).toBeCalled();
    expect(mockCoc.getClan.mock.calls[0][0]).toMatchSnapshot();
    expect(mockDb.get).toBeCalled();
    expect(mockDb.get.mock.calls[0][0].Key).toMatchSnapshot();
    expect(mockDb.query).toBeCalled();
    expect(mockDb.query.mock.calls[0][0].KeyConditionExpression).toMatchSnapshot();
    expect(mockDb.query.mock.calls[1][0].KeyConditionExpression).toMatchSnapshot();
    expect(mockDb.put).toBeCalled();
    expect(mockDb.put.mock.calls[0][0].Item).toMatchSnapshot();
    expect(mockDb.delete).toBeCalled();
    expect(mockDb.delete.mock.calls[0][0].Key).toMatchSnapshot();
  }));
});

describe('given server clan is already leader verified', () => {
  it.effect('when registering', E.fn(function* () {
    mockDb.get
      .mockReturnValueOnce(E.succeed({Item: TestDataServer}));

    mockDb.query
      .mockReturnValueOnce(E.succeed({Items: [TestDataUser, TestDataUserPlayer]}))
      .mockReturnValueOnce(E.succeed({Items: [TestDataServerClanLeaderVerified]}));

    mockCoc.getClan
      .mockReturnValueOnce(E.succeed({
        tag        : 'clan',
        name       : 'ClanName',
        description: 'ClanDescription',
        members    : [{
          tag : 'player',
          role: 'elder',
        }],
      }));

    const actual = yield* pipe(
      ServerClanRegistry.register({
        caller_id   : 'user',
        caller_roles: ['admin'],
        guild_id    : 'guild',
        clan_tag    : 'clan',
        payload     : {
          countdown: 'countdown',
        },
      }),
      E.provide(mockDbLayer),
      E.provide(mockCocLayer),
      E.catchAll((cause) => E.succeed(cause)),
    );

    expect(actual).toMatchInlineSnapshot(`[RegistryAdminError: Your member player links cannot override the current verification level of this clan.]`);
    expect(mockCoc.getClan).toBeCalled();
    expect(mockCoc.getClan.mock.calls[0][0]).toMatchSnapshot();
    expect(mockDb.get).toBeCalled();
    expect(mockDb.get.mock.calls[0][0].Key).toMatchSnapshot();
    expect(mockDb.query).toBeCalled();
    expect(mockDb.query.mock.calls[0][0].KeyConditionExpression).toMatchSnapshot();
    expect(mockDb.query.mock.calls[1][0].KeyConditionExpression).toMatchSnapshot();
    expect(mockDb.put).not.toBeCalled();
    expect(mockDb.delete).not.toBeCalled();
  }));
});
