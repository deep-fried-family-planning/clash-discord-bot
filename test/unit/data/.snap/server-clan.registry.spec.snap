// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`given caller has no verified accounts in clan > when registering 2`] = `"clan"`;

exports[`given caller has no verified accounts in clan > when registering 3`] = `
{
  "pk": "s-guild",
  "sk": "now",
}
`;

exports[`given caller has no verified accounts in clan > when registering 4`] = `"pk = :pk"`;

exports[`given caller has no verified accounts in clan > when registering 5`] = `"gsi_clan_tag = :gsi_clan_tag"`;

exports[`given caller has no verified accoutns in clan > when registering 2`] = `
{
  "pk": "s-guild",
  "sk": "now",
}
`;

exports[`given caller is not admin > when registering 2`] = `
{
  "pk": "s-guild",
  "sk": "now",
}
`;

exports[`given caller user is not registered > when registering 2`] = `
{
  "pk": "s-guild",
  "sk": "now",
}
`;

exports[`given caller user is not registered > when registering 3`] = `"pk = :pk"`;

exports[`given server clan is already leader verified > when registering 2`] = `"clan"`;

exports[`given server clan is already leader verified > when registering 3`] = `
{
  "pk": "s-guild",
  "sk": "now",
}
`;

exports[`given server clan is already leader verified > when registering 4`] = `"pk = :pk"`;

exports[`given server clan is already leader verified > when registering 5`] = `"gsi_clan_tag = :gsi_clan_tag"`;

exports[`given server clan is already leader verified > when registering 6`] = `
{
  "_tag": "ServerClan",
  "countdown": "countdown",
  "description": "ClanDescription",
  "gsi_clan_tag": "c-clan",
  "gsi_server_id": "s-guild",
  "name": "ClanName",
  "pk": "s-guild",
  "select": {
    "label": "ClanName",
    "value": "c-clan",
  },
  "sk": "c-clan",
  "verification": 3,
  "version": 0,
}
`;

exports[`given server clan is already leader verified > when registering 7`] = `
{
  "pk": "s-guild2",
  "sk": "c-clan",
}
`;

exports[`given server clan is already registered with elder verification > when registering 2`] = `"clan"`;

exports[`given server clan is already registered with elder verification > when registering 3`] = `
{
  "pk": "s-guild",
  "sk": "now",
}
`;

exports[`given server clan is already registered with elder verification > when registering 4`] = `"pk = :pk"`;

exports[`given server clan is already registered with elder verification > when registering 5`] = `"gsi_clan_tag = :gsi_clan_tag"`;

exports[`given server clan is already registered with elder verification > when registering 6`] = `
{
  "_tag": "ServerClan",
  "countdown": "countdown",
  "description": "ClanDescription",
  "gsi_clan_tag": "c-clan",
  "gsi_server_id": "s-guild",
  "name": "ClanName",
  "pk": "s-guild",
  "select": {
    "label": "ClanName",
    "value": "c-clan",
  },
  "sk": "c-clan",
  "verification": 3,
  "version": 0,
}
`;

exports[`given server clan is already registered with elder verification to a different server > when registering 2`] = `"clan"`;

exports[`given server clan is already registered with elder verification to a different server > when registering 3`] = `
{
  "pk": "s-guild",
  "sk": "now",
}
`;

exports[`given server clan is already registered with elder verification to a different server > when registering 4`] = `"pk = :pk"`;

exports[`given server clan is already registered with elder verification to a different server > when registering 5`] = `"gsi_clan_tag = :gsi_clan_tag"`;

exports[`given server clan is already registered with elder verification to a different server > when registering 6`] = `
{
  "_tag": "ServerClan",
  "countdown": "countdown",
  "description": "ClanDescription",
  "gsi_clan_tag": "c-clan",
  "gsi_server_id": "s-guild",
  "name": "ClanName",
  "pk": "s-guild",
  "select": {
    "label": "ClanName",
    "value": "c-clan",
  },
  "sk": "c-clan",
  "verification": 3,
  "version": 0,
}
`;

exports[`given server clan is already registered with elder verification to a different server > when registering 7`] = `
{
  "pk": "s-guild2",
  "sk": "c-clan",
}
`;

exports[`when registering a new server clan 2`] = `"clan"`;

exports[`when registering a new server clan 3`] = `
{
  "pk": "s-guild",
  "sk": "now",
}
`;

exports[`when registering a new server clan 4`] = `"pk = :pk"`;

exports[`when registering a new server clan 5`] = `"gsi_clan_tag = :gsi_clan_tag"`;

exports[`when registering a new server clan 6`] = `
{
  "_tag": "ServerClan",
  "countdown": "countdown",
  "description": "ClanDescription",
  "gsi_clan_tag": "c-clan",
  "gsi_server_id": "s-guild",
  "name": "ClanName",
  "pk": "s-guild",
  "select": {
    "label": "ClanName",
    "value": "c-clan",
  },
  "sk": "c-clan",
  "verification": 3,
  "version": 0,
}
`;
